component/otelcol/processor/probabilistic_sampler/probabilistic_sampler.go

```diff
@@ -0,0 +1,91 @@
+// Package probabilistic_sampler provides an otelcol.processor.probabilistic_sampler component.
+package probabilistic_sampler
+
+import (
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/otelcol"
+	"github.com/grafana/agent/component/otelcol/processor"
+	psp "github.com/open-telemetry/opentelemetry-collector-contrib/processor/probabilisticsamplerprocessor"
+	otelcomponent "go.opentelemetry.io/collector/component"
+	otelextension "go.opentelemetry.io/collector/extension"
+)
+
+const (
+	defaultAttributeSource = "traceID"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "otelcol.processor.probabilistic_sampler",
+		Args:    Arguments{},
+		Exports: otelcol.ConsumerExports{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			fact := psp.NewFactory()
+			return processor.New(opts, fact, args.(Arguments))
+		},
+	})
+}
+
+// Arguments configures the otelcol.processor.probabilistic_sampler component.
+
+type Arguments struct {
+	SamplingPercentage float32 `river:"sampling_percentage,attr,optional"`
```

Comment:
I think `sampling_percentage` is not optional? The otel docs list it as required:
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.80.0/processor/probabilisticsamplerprocessor/README.md

Reply:
yeah, it is listed as required, but actually, it isn't. If `sampling_percentage` is not specified then it is 0 by default. I will prepare PR upstream to fix README